radixdlt/babylon-wallet-ios
# [ABW-2409] Import Many Olympia Accounts

Jira ticket: [ABW-2409](https://radixdlt.atlassian.net/browse/ABW-2409)

## Description
This PR refactors the validation and migration logic in ImportOlympiaLedgerAccountsAndFactorSources, to prevent it from failing when importing many (20 or more) Olympia hardware accounts.

## How to test
- Create at least 20 Ledger accounts using the Olympia wallet
- Export them, creating a QR code
- Scan the QR in the wallet to import them
- The flow should finish normally, and not get stuck after validating all the accounts

## Video
After:

https://github.com/radixdlt/babylon-wallet-ios/assets/123396602/57695182-8059-42f1-a584-b27b9c6c7f3a

## PR submission checklist
